src/number_theory/wilson.lean

```diff
@@ -0,0 +1,58 @@
+/-
+Copyright (c) 2022 John Nicol. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: John Nicol
+-/
+import number_theory.legendre_symbol.gauss_eisenstein_lemmas
+
+/-!
+# Wilson's theorem.
+
+This file contains a proof of Wilson's theorem.
+
+The heavy lifting is mostly done by the previous `wilsons_lemma`,
+but here we also prove the other logical direction.
+
+This could be generalized to similar results about finite abelian groups.
+
+## References
+
+* [Wilson's Theorem](https://en.wikipedia.org/wiki/Wilson%27s_theorem)
+
+-/
+
+open_locale nat
+
+namespace nat
+
+/-- For `n > 1`, `(n-1)!` is congruent to `-1` modulo `n` only if n is prime. --/
+private lemma wilsons_theorem_only_if_direction
+  (n : ℕ) (h : ((n - 1)! : zmod n) = -1) (h1 : 1 < n) : prime n :=
```

Comment:
```suggestion
  {n : ℕ} (h : ((n - 1)! : zmod n) = -1) (h1 : 1 < n) : prime n :=
```

Reply:
Yes, making `n` implicit in this lemma means you should omit `n` on line 50 when invoking this lemma.

Reply:
Yes, this is what the implicit binder `{n}` means. This is a good thing to review in TPiL if you're not familiar with it!